spec/controllers/event_controller_spec.rb

```diff
@@ -0,0 +1,104 @@
+require 'rails_helper'
+
+describe 'EventController', type: :controller do
+  render_views
+
+  before :each do
+    @controller = EventsController.new
+  end
+
+  describe '#index' do
+    it 'assigns @events' do
+      event = Event.create(title: 'Valid title')
+      get :index
+      expect(assigns(:events)).to eq([event])
+    end
+    it 'properly renders' do
+      get :index
+      expect(response).to render_template(:index)
+    end
+  end
+
+  describe '#new' do
+    before :each do
+      get :new
+    end
+    it 'assigns a new Event to @event' do
+      expect(assigns(:event)).to be_a_new(Event)
+    end
+  end
+
+  describe '#create' do
+    it 'redirects to event' do
+      post :create, event: FactoryGirl.attributes_for(:event)
+      expect(response).to redirect_to(Event.last)
+    end
+    it 'renders new page on failed save' do
+      post :create, event: FactoryGirl.attributes_for(:event, title: nil)
+      expect(response).to render_template(:new)
+    end
+  end
+
+  describe '#edit' do
+    before :each do
+      @event = Event.create(title: 'event title')
+      get :edit, id: @event.id
+    end
+    it 'properly renders' do
+      expect(response).to render_template(:edit)
+    end
+  end
+
+  describe '#update' do
```

Comment:
Or the vase where they update and don't actually change the contents of the event?